src/bin/nydus-image/main.rs

```diff
@@ -1041,6 +1050,12 @@ impl Command {
         build_ctx.set_chunk_size(chunk_size);
         build_ctx.set_batch_size(batch_size);
 
+        let blob_cache_generator = match blob_cache_storage {
+            Some(storage) => Some(Mutex::new(BlobCacheGenerator::new(storage)?)),
+            None => None,
+        };
+        build_ctx.blob_cache_generator = blob_cache_generator;
```

Comment:
```
build_ctx.blob_cache_generator = blob_cache_storage.map(|v| {Mutex::new(BlobCacheGeneratorNew(v))});
```

Reply:
BlobCacheGenerator::new(storage) returns a Result, it can't return the error directly inside the closure